### PR TITLE
Update CA1036 messaging.

### PR DESCRIPTION
**Pull request by 333fred**
Fixes https://github.com/dotnet/roslyn-analyzers/issues/1396. Tagging @mavasani @ivanbasov for review.

### CODE REVIEW

**Comment by 333fred** [comment]
Ping.

File: src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/OverrideMethodsOnComparableTypes.cs [hunk 1]
```diff
@@ -73,12 +98,70 @@ private static void AnalyzeSymbol(INamedTypeSymbol namedTypeSymbol, INamedTypeSy
...
+                {
+                    addDiagnostic(namedTypeSymbol.CreateDiagnostic(RuleOperator, namedTypeSymbol.Name, comparisonOperatorsString));
+                }
+            }
+        }
+
+        private static string GetNeededComparisonOperators(INamedTypeSymbol symbol)
+        {
+            bool first = true;
+            StringBuilder sb = new StringBuilder();
```
Comment (mavasani) [inline src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/OverrideMethodsOnComparableTypes.cs]: Minor: initialize to null, add `sb = sb ?? new StringBuilder();` at the start of append, and add a null check at return.

File: src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/OverrideMethodsOnComparableTypes.cs [hunk 2]
```diff
@@ -22,20 +23,44 @@ public sealed class OverrideMethodsOnComparableTypesAnalyzer : DiagnosticAnalyze
...
+                                                                                  s_localizableTitle,
+                                                                                  s_localizableMessageOperators,
+                                                                                  DiagnosticCategory.Design,
+                                                                                  DiagnosticHelpers.DefaultDiagnosticSeverity,
+                                                                                  isEnabledByDefault: DiagnosticHelpers.EnabledByDefaultForVsixAndNuget,
+                                                                                  description: s_localizableDescription,
+                                                                                  helpLinkUri: "https://docs.microsoft.com/visualstudio/code-quality/ca1036-override-methods-on-comparable-types",
+                                                                                  customTags: WellKnownDiagnosticTags.Telemetry);
+
+
```
Comment (ivanbasov) [inline src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/OverrideMethodsOnComparableTypes.cs]: nit: remove extra line

Comment (333fred) [reply src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/OverrideMethodsOnComparableTypes.cs]: Fixed.

**Pull request merged**